lesson02.md

```diff
@@ -0,0 +1,440 @@
+<!-- .slide: id="lesson2" -->
+
+# Basic Frontend - Spring 2022
+
+Lesson 2, Thursday, 2022-03-17
+
+---
+
+### Recap
+
+What did we talk about last lesson?
+
+---
+
+### Recap: data types
+
+Number:
+```js
+42
+-3.1415
+```
+
+String:
+```js
+"John"
+'Berlin'
+```
+
+Boolean:
+```js
+true
+false
+```
+
+---
+
+### Recap operators
+
+| Group | Operators | Example |
+| ----- | --------- | ------- |
+| Numerical Operators | `+ - * / **` | `5 + 4 * 3`<br/>`7 / 2 - 2`<br/>`"Hello" + " World"` |
+| Comparison Operators | `=== !== < > <= >=` | `30 !== 25`<br/>`20 >= 18` |
+
+---
+
+### Try it out
+
+Now, try to enter some data to the JavaScript console, but think of the data type first:
+
+* Your name
+* Your shoe size
+* Wether you think it's ok to put pineapple on pizza
+
+---
+
+### Try it out: Solution
+
+```js
+"Carlo"
+45
+false
+```
+
+---
+
+### Comparison operators
+
+Use JavaScript to check if:
+
+* you are older than me (I'm 44)
+* your first name is the same as mine
+
+---
+
+### Solution
+
+```js
+26 > 44                // false - I am older
+"Harald" === "Owen"    // false - not the same first name
+```
+
+---
+
+### Summary
+
+So far, we learned how to enter data and how to use operators to derive new data.
+
+But when reading the code we wrote, we won't quite understand it later:
+
+```js
+4*8
+32/2
+```
+
+What's that? How can we make it clearer what that means?
+
+---
+
+<!-- .slide: id="variables" -->
+
+# Variables
+
+---
+
+### Variable
+
+A pointer to value.
+
+```js
+let price = 5;
+let name = "John";
+```
+
+![Variable](images/variables.jpg)
+<!-- .element: style="text-align:center" -->
+
+---
+
+### Examples:
+
+```js
+let priceCoffee = 2;
+let priceCappuccino = 3;
+let customer = "John";
+let greeting = "Hello";
+let likesCarlo = true;
+let hasKids = false;
+```
+
+Can you guess how we can define a variable?
+
+---
+
+### How do we define a variable?
+
+We write:
+* `let`
+* the variable name
+* `=`
+* the value we want
+* `;`
+
+---
+
+### Variables are variable
+
+You can change the value of a variable using the assignment operator (`=`):
+
+```js
+let temperature = 24;   // define a variable, initialize it to 24
+temperature = 25;       // assign a new value (25) to the variable
```

Comment:
I might miss it, but I think we don't mention anywhere that you can't initialise two vars with the same name. as:
let haraldAge = 20
let haraldAge = 21

Instead of
let haraldAge = 20
     haraldAge = 21